design-proposals/klip-54-optimized-range-pull-queries.md

```diff
@@ -0,0 +1,52 @@
+# KLIP 53 - Optimized Range Pull Queries 
+
+**Author**: Patrick Stuedi (@pstuedi) | 
+**Release Target**: 0.22 | 
+**Status**: _In Discussion_ | 
+**Discussion**: https://github.com/confluentinc/ksql/pull/7993
+
+**tl;dr:** For range pull queries on tables, use the range interface provided by the state store for retrieving records instead of doing a table scan.
+           
+## Motivation and background
+
+Currently, range queries on primary keys (e.g., WHERE colX < T) are implemented using a table scan followed by a filtering operation, which is inefficient. Recently, range scan capabilities were added to Kafka streams (#4064). Following that work, we should optimize the execution of range queries in ksqlDB by directly leveraging the dedicated range query interface to improve I/O efficiency and performance. 
+
+## What is in scope
```

Comment:
I think this is a good place to discuss the implications of serdes on range scans, and how we eventually plan to support non-string range scans

Reply:
Added a paragraph in the "What is not in this scope" about the String restriction and how we can support binary types in the future.